### PR TITLE
Fix for Apple Silicon

### LINKED ISSUE #9: Failed to build on Apple Silicon
macOS Big Sur 11.0 (20A2411) Mac mini (M1, 2020)

```
$ rustc -V
rustc 1.50.0-nightly (1c389ffef 2020-11-24)

$ cargo build --tests -vv
       Fresh cc v1.0.65
       Fresh remove_dir_all v0.5.3
       Fresh lazy_static v1.4.0
       Fresh libc v0.2.80
       Fresh rand v0.4.6
       Fresh libloading v0.5.2
       Fresh tempdir v0.3.7
   Compiling fs-swap v0.2.4 (/Users/xiliangchen/projects/fs-swap)
     Running `CARGO=/Users/xiliangchen/.rustup/toolchains/nightly-aarch64-apple-darwin/bin/cargo CARGO_CRATE_NAME=fs_swap CARGO_MANIFEST_DIR=/Users/xiliangchen/projects/fs-swap CARGO_PKG_AUTHORS='debris <marek.kotewicz@gmail.com>' CARGO_PKG_DESCRIPTION='Cross-platform implementation of path swap' CARGO_PKG_HOMEPAGE='https://github.com/debris/fs-swap' CARGO_PKG_LICENSE=MIT CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=fs-swap CARGO_PKG_REPOSITORY='' CARGO_PKG_VERSION=0.2.4 CARGO_PKG_VERSION_MAJOR=0 CARGO_PKG_VERSION_MINOR=2 CARGO_PKG_VERSION_PATCH=4 CARGO_PKG_VERSION_PRE='' CARGO_PRIMARY_PACKAGE=1 DYLD_FALLBACK_LIBRARY_PATH='/Users/xiliangchen/projects/fs-swap/target/debug/deps:/Users/xiliangchen/.rustup/toolchains/nightly-aarch64-apple-darwin/lib:/Users/xiliangchen/.rustup/toolchains/nightly-aarch64-apple-darwin/lib:/Users/xiliangchen/lib:/usr/local/lib:/usr/lib' rustc --crate-name fs_swap src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 --test -C metadata=113eb5d54c650c66 -C extra-filename=-113eb5d54c650c66 --out-dir /Users/xiliangchen/projects/fs-swap/target/debug/deps -C incremental=/Users/xiliangchen/projects/fs-swap/target/debug/incremental -L dependency=/Users/xiliangchen/projects/fs-swap/target/debug/deps --extern lazy_static=/Users/xiliangchen/projects/fs-swap/target/debug/deps/liblazy_static-522329360d70356d.rlib --extern libc=/Users/xiliangchen/projects/fs-swap/target/debug/deps/liblibc-4048b54428fda517.rlib --extern libloading=/Users/xiliangchen/projects/fs-swap/target/debug/deps/liblibloading-541f2263c7fbefc0.rlib --extern tempdir=/Users/xiliangchen/projects/fs-swap/target/debug/deps/libtempdir-fa4b59932425d62a.rlib -L native=/Users/xiliangchen/projects/fs-swap/target/debug/build/libloading-c9dc113183cafbfb/out`
LLVM ERROR: Unsupported calling convention.
error: could not compile `fs-swap`

Caused by:
  process didn't exit successfully: `CARGO=/Users/xiliangchen/.rustup/toolchains/nightly-aarch64-apple-darwin/bin/cargo CARGO_CRATE_NAME=fs_swap CARGO_MANIFEST_DIR=/Users/xiliangchen/projects/fs-swap CARGO_PKG_AUTHORS='debris <marek.kotewicz@gmail.com>' CARGO_PKG_DESCRIPTION='Cross-platform implementation of path swap' CARGO_PKG_HOMEPAGE='https://github.com/debris/fs-swap' CARGO_PKG_LICENSE=MIT CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=fs-swap CARGO_PKG_REPOSITORY='' CARGO_PKG_VERSION=0.2.4 CARGO_PKG_VERSION_MAJOR=0 CARGO_PKG_VERSION_MINOR=2 CARGO_PKG_VERSION_PATCH=4 CARGO_PKG_VERSION_PRE='' CARGO_PRIMARY_PACKAGE=1 DYLD_FALLBACK_LIBRARY_PATH='/Users/xiliangchen/projects/fs-swap/target/debug/deps:/Users/xiliangchen/.rustup/toolchains/nightly-aarch64-apple-darwin/lib:/Users/xiliangchen/.rustup/toolchains/nightly-aarch64-apple-darwin/lib:/Users/xiliangchen/lib:/usr/local/lib:/usr/lib' rustc --crate-name fs_swap src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 --test -C metadata=113eb5d54c650c66 -C extra-filename=-113eb5d54c650c66 --out-dir /Users/xiliangchen/projects/fs-swap/target/debug/deps -C incremental=/Users/xiliangchen/projects/fs-swap/target/debug/incremental -L dependency=/Users/xiliangchen/projects/fs-swap/target/debug/deps --extern lazy_static=/Users/xiliangchen/projects/fs-swap/target/debug/deps/liblazy_static-522329360d70356d.rlib --extern libc=/Users/xiliangchen/projects/fs-swap/target/debug/deps/liblibc-4048b54428fda517.rlib --extern libloading=/Users/xiliangchen/projects/fs-swap/target/debug/deps/liblibloading-541f2263c7fbefc0.rlib --extern tempdir=/Users/xiliangchen/projects/fs-swap/target/debug/deps/libtempdir-fa4b59932425d62a.rlib -L native=/Users/xiliangchen/projects/fs-swap/target/debug/build/libloading-c9dc113183cafbfb/out` (exit code: 101)
```

### PR DESCRIPTION
**Pull request by xlc**
Closes #9

### CODE REVIEW

File: src/platform/macos.rs [hunk 1]
```diff
@@ -14,7 +14,7 @@ lazy_static! {
 	};
 }
 
-
+#[cfg(target_arch = "x86_64")]
 extern "stdcall" {
```
Comment (ordian) [inline src/platform/macos.rs:18]: I wonder why is this `stdcall`, I thought this is for windows. And macOS uses System V, no? Does extern "C" work?

Comment (niklasad1) [reply src/platform/macos.rs:18]: Good question, according to https://doc.rust-lang.org/stable/reference/items/external-blocks.html#abi  `System V` is the default for macOS but it has worked in the past AFAIK.

//cc @debris

Comment (ordian) [reply src/platform/macos.rs:18]: Hmm, maybe it worked because this function was not called at all?

**Comment by debris** [comment]
thanks 🙌 

I added `paritytech:core-devs` to the owners on [crates.io/fs-swap](https://crates.io/crates/fs-swap). Feel free to release the new version

**Comment by ordian** [comment]
Thanks, opened https://github.com/paritytech/fs-swap/pull/1.

**Pull request merged**